PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/selection/GraphBubbleDescriptionBuilder.java

```diff
@@ -0,0 +1,112 @@
+package nl.tudelft.pl2016gr2.gui.view.selection;
+
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Node;
+import javafx.scene.layout.Pane;
+import nl.tudelft.pl2016gr2.model.graph.nodes.GraphNode;
+import nl.tudelft.pl2016gr2.model.graph.nodes.IndelBubble;
+import nl.tudelft.pl2016gr2.model.graph.nodes.PhyloBubble;
+import nl.tudelft.pl2016gr2.model.graph.nodes.PointMutationBubble;
+import nl.tudelft.pl2016gr2.model.graph.nodes.SequenceNode;
+import nl.tudelft.pl2016gr2.model.graph.nodes.StraightSequenceBubble;
+import nl.tudelft.pl2016gr2.visitor.NodeVisitor;
+
+/**
+ * This class can be used to build {@link ISelectionInfo} for a {@link GraphNode}.
+ *
+ * <p>
+ * The internal logic uses the visitor-pattern to provide a fitting result for each
+ * given {@link GraphNode}.
+ * </p>
+ * <p>
+ * Please see {@link #buildInfo(GraphNode)} when you want to use this class.
+ * </p>
+ *
+ */
+public class GraphBubbleDescriptionBuilder implements NodeVisitor {
```

Comment:
Not sure this is the original idea behind `NodeVisitor` objects, but I guess the pattern is very flexible?